File: Example_openGL3/hello.py
import sys
import numpy as np

from OpenGL.GL import *
from OpenGL.GL import shaders
from OpenGL.GLUT import *

vao = None;
vbo = None;
shaderProgram = None;

def readShaderFile(filename):
	with open('shader/' + filename, 'r') as myfile:
		return myfile.read()

def init():
	global shaderProgram
	global vao
	global vbo
	
	glClearColor(0, 0, 0, 0);
	
	vertex_code = readShaderFile('hello.vp')
	fragment_code = readShaderFile('hello.fp')

	# compile shaders and program
	vertexShader = shaders.compileShader(vertex_code, GL_VERTEX_SHADER)
	fragmentShader = shaders.compileShader(fragment_code, GL_FRAGMENT_SHADER)
	shaderProgram = shaders.compileProgram(vertexShader, fragmentShader)
	
	# Create and bind the Vertex Array Object
	vao = GLuint(0)
	glGenVertexArrays(1, vao)
	glBindVertexArray(vao)

	# Create and bind the Vertex Buffer Object
	vertices = np.array([[0, 1, 0], [-1, -1, 0], [1, -1, 0]], dtype='f')
	vbo = glGenBuffers(1)
	glBindBuffer(GL_ARRAY_BUFFER, vbo)
	glBufferData(GL_ARRAY_BUFFER, vertices, GL_STATIC_DRAW)

	glVertexAttribPointer(0, 3, GL_FLOAT, False, 0, None)  # first 0 is the location in shader
	glBindAttribLocation(shaderProgram, 0, 'vertexPosition')  # name of attribute in shader
	glEnableVertexAttribArray(0);  # 0=location do atributo, tem que ativar todos os atributos inicialmente sao desabilitados por padrao
	
	# Note that this is allowed, the call to glVertexAttribPointer registered VBO
	# as the currently bound vertex buffer object so afterwards we can safely unbind
	glBindBuffer(GL_ARRAY_BUFFER, 0);
	# Unbind VAO (it's always a good thing to unbind any buffer/array to prevent strange bugs)
	glBindVertexArray(0);

def display():
	global shaderProgram
	global vao
	
	glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT)
	
	# load everthing back
	glUseProgram(shaderProgram)
	glBindVertexArray(vao)
	glBindBuffer(GL_ARRAY_BUFFER, vbo)
	
	# glDrawArrays( mode , first, count)
	glDrawArrays(GL_TRIANGLES, 0, 3)
	
	#clean things up
	glBindBuffer(GL_ARRAY_BUFFER, 0)
	glBindVertexArray(0)
	glUseProgram(0)
	
	glutSwapBuffers()  # necessario para windows!

def reshape(width, height):
	glViewport(0, 0, width, height)

if __name__ == '__main__':
	glutInit()
	glutInitContextVersion(3, 0)
	glutInitContextProfile(GLUT_CORE_PROFILE);
	glutInitDisplayMode(GLUT_DOUBLE | GLUT_RGB | GLUT_DEPTH)
	
	glutInitWindowSize(640, 480);
	glutCreateWindow(b'Hello world!')
	
	glutReshapeFunc(reshape)
	glutDisplayFunc(display)

	init()
	
	glutMainLoop()

File: Example_openGL3/Hello.java

//need lib jogamp
import java.nio.FloatBuffer;

import javax.swing.JFrame;

import com.jogamp.common.nio.Buffers;
import com.jogamp.opengl.GL;
import com.jogamp.opengl.GL3;
import com.jogamp.opengl.GLAutoDrawable;
import com.jogamp.opengl.GLCapabilities;
import com.jogamp.opengl.GLEventListener;
import com.jogamp.opengl.GLException;
import com.jogamp.opengl.GLProfile;
import com.jogamp.opengl.awt.GLJPanel;
import com.jogamp.opengl.util.GLBuffers;
import com.jogamp.opengl.util.glsl.ShaderCode;
import com.jogamp.opengl.util.glsl.ShaderProgram;

public class Hello implements GLEventListener {

	private ShaderProgram shaderProgram;
	private ShaderCode vertexShader;
	private ShaderCode fragmentShader;

	private int[] vao = new int[1];
	private int[] vbo = new int[1];

	@Override
	public void init(GLAutoDrawable canvas) {
		GL3 gl = canvas.getGL().getGL3();
		gl.glClearColor(0f, 0f, 0f, 0f);
		// Create and compile our GLSL program from the shaders
		createShaders(gl);

		// Create and Bind the Vertex Array Object first, then bind and set vertex buffer(s) and attribute pointer(s).
		gl.glGenVertexArrays(1, vao, 0);
		gl.glBindVertexArray(vao[0]);

		// Set up vertex data (and buffer(s)) and attribute pointers
		float vertexData[] = { -1.0f, -1.0f, 0.0f, 1.0f, -1.0f, 0.0f, 0.0f, 1.0f, 0.0f };
		FloatBuffer vertexDataBuffer = GLBuffers.newDirectFloatBuffer(vertexData);
		gl.glGenBuffers(1, vbo, 0);
		gl.glBindBuffer(GL.GL_ARRAY_BUFFER, vbo[0]);
		gl.glBufferData(GL.GL_ARRAY_BUFFER, Buffers.SIZEOF_FLOAT * vertexDataBuffer.limit(), vertexDataBuffer, GL.GL_STATIC_DRAW);

		// 0=location do atributo no shader
		// penultimo zero pode ser 3 * Float.BYTES
		gl.glVertexAttribPointer(0, 3, GL.GL_FLOAT, false, 0, 0);// first 0 is the location in shader
		// tem que ativar todos os atributos inicialmente sao desabilitados por padrao
		gl.glEnableVertexAttribArray(0);
		gl.glBindAttribLocation(shaderProgram.program(), 0, "vertexPosition");// name of attribute in shader
		// Note that this is allowed, the call to glVertexAttribPointer registered VBO
		// as the currently bound vertex buffer object so afterwards we can safely unbind
		gl.glBindBuffer(GL.GL_ARRAY_BUFFER, 0);
		// Unbind VAO (it's always a good thing to unbind any buffer/array to prevent strange bugs)
		gl.glBindVertexArray(0);
	}

	private void createShaders(GL3 gl) {
		vertexShader = ShaderCode.create(gl, GL3.GL_VERTEX_SHADER, getClass(), "shader", null, "hello", false);
		fragmentShader = ShaderCode.create(gl, GL3.GL_FRAGMENT_SHADER, getClass(), "shader", null, "hello", false);

		if (!vertexShader.compile(gl, System.err))
			throw new GLException("Couldn't compile shader: " + vertexShader);
		if (!fragmentShader.compile(gl, System.err))
			throw new GLException("Couldn't compile shader: " + fragmentShader);

		shaderProgram = new ShaderProgram();
		shaderProgram.add(gl, vertexShader, System.err);
		shaderProgram.add(gl, fragmentShader, System.err);
		shaderProgram.link(gl, System.out);
	}

	@Override
	public void display(GLAutoDrawable canvas) {
		GL3 gl = canvas.getGL().getGL3();
		gl.glClear(GL.GL_DEPTH_BUFFER_BIT | GL.GL_COLOR_BUFFER_BIT);

		// load everthing back
		gl.glUseProgram(shaderProgram.program());
		gl.glBindVertexArray(vao[0]);
		gl.glBindBuffer(GL.GL_ARRAY_BUFFER, vbo[0]);
		// Draw our first triangle
		gl.glDrawArrays(GL.GL_TRIANGLES, 0, 3);

		// clean things up
		gl.glUseProgram(0);
	}

	@Override
	public void reshape(GLAutoDrawable canvas, int x, int y, int width, int height) {
		System.out.println("reshape");

		GL3 gl3 = canvas.getGL().getGL3();
		gl3.glViewport(x, y, width, height);
	}

	@Override
	public void dispose(GLAutoDrawable drawable) {
		System.out.println("dispose");

		GL3 gl3 = drawable.getGL().getGL3();
		gl3.glDeleteBuffers(1, vbo, 0);
		gl3.glDeleteVertexArrays(1, vao, 0);

		gl3.glDeleteProgram(shaderProgram.id());
		gl3.glDeleteShader(vertexShader.id());
		gl3.glDeleteShader(fragmentShader.id());
	}

	public static void main(String[] args) {
		JFrame app = new JFrame("Minimal OpenGL 3 with shader");
		app.setSize(640, 480);
		// this.setResizable(false);
		app.setLocationRelativeTo(null);
		app.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

		GLProfile profile = GLProfile.get(GLProfile.GL3);

		GLJPanel canvas = new GLJPanel(new GLCapabilities(profile));
		canvas.addGLEventListener(new Hello());
		app.getContentPane().add(canvas);

		canvas.requestFocusInWindow();
		app.setVisible(true);
	}
}
